aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -0,0 +1,2434 @@
+    /// On each `add_stake` operation any resulted fee is used to buy active shares for this shareholder.
+    /// First synchronization after this epoch ends will distribute accumulated fees to the rest of the pool as refunds.
+    const NULL_SHAREHOLDER: address = @0x0;
+
+    /// Capability that represents ownership over privileged operations on the underlying stake pool.
+    struct DelegationPoolOwnership has key, store {
+        /// equal to address of the resource account owning the stake pool
+        pool_address: address,
+    }
+
+    struct ObservedLockupCycle has copy, drop, store {
+        index: u64,
+    }
+
+    struct DelegationPool has key {
+        // Shares pool of `active` + `pending_active` stake
+        active_shares: pool_u64::Pool,
+        // Index of current observed lockup cycle on the delegation pool since its creation
+        observed_lockup_cycle: ObservedLockupCycle,
+        // Shares pools of `inactive` stake on each ended OLC and `pending_inactive` stake on the current one.
+        // Tracks shares of delegators who requested withdrawals in each OLC
+        inactive_shares: Table<ObservedLockupCycle, pool_u64::Pool>,
+        // Mapping from delegator address to the OLC of its pending withdrawal if having one
+        pending_withdrawals: Table<address, ObservedLockupCycle>,
+        // Signer capability of the resource account owning the stake pool
+        stake_pool_signer_cap: account::SignerCapability,
+        // Total (inactive) coins on the shares pools over all ended OLCs
+        total_coins_inactive: u64,
+        // Commission fee paid to the node operator out of pool rewards
+        operator_commission_percentage: u64,
+
+        // The events emitted by stake-management operations on the delegation pool
+        add_stake_events: EventHandle<AddStakeEvent>,
+        reactivate_stake_events: EventHandle<ReactivateStakeEvent>,
+        unlock_stake_events: EventHandle<UnlockStakeEvent>,
+        withdraw_stake_events: EventHandle<WithdrawStakeEvent>,
+        distribute_commission_events: EventHandle<DistributeCommissionEvent>,
+    }
+
+    struct AddStakeEvent has drop, store {
+        pool_address: address,
+        delegator_address: address,
+        amount_added: u64,
+        add_stake_fee: u64,
+    }
+
+    struct ReactivateStakeEvent has drop, store {
+        pool_address: address,
+        delegator_address: address,
+        amount_reactivated: u64,
+    }
+
+    struct UnlockStakeEvent has drop, store {
+        pool_address: address,
+        delegator_address: address,
+        amount_unlocked: u64,
+    }
+
+    struct WithdrawStakeEvent has drop, store {
+        pool_address: address,
+        delegator_address: address,
+        amount_withdrawn: u64,
+    }
+
+    struct DistributeCommissionEvent has drop, store {
+        pool_address: address,
+        operator: address,
+        commission_active: u64,
+        commission_pending_inactive: u64,
+    }
+
```

Comment:
Can you add view functions for the following?
1. The number of shareholders in the active pool share. This would be useful to use as a rough total number of delegators (there might still be some on the way out in the last inactive shares pool, but that's okay as it's not easy to deduplicate)
2. The total amount of active stake in the delegation pool.

Reply:
2. `stake::get_stake` could be used instead if marked as `view`, otherwise would be more complete to proxy it by defining `delegation_pool:get_stake` which returns all individual stake states not just the active one?